oz/Guest.py

```diff
@@ -1115,14 +1115,22 @@ def _wait_for_guest_boot(self, libvirt_dom):
             self.addr = None
             self.data = ''
 
+            # The sclplm console used on s390x is connected to the sclp console
+            # and thus it will receive all the boot messages.  Hence, read more
+            # data to drain all the boot messages.
```

Comment:
Is this just a performance optimization?  While I'm not really against this, I'm wondering why reading 100 bytes at a time isn't good enough to eventually drain all of the messages.

Reply:
Not really: as the comment says, basically here we read all the boot messages, which for a simple recent Fedora/RHEL guest this accounts to 30K or so of data.  Reading that amount in 100 bytes chunks every 10 seconds does not allow to wait for the guest boot within the default amount (300 seconds).

The other alternative here would be drain the socket every time `_boot_cb` is invoked.

Reply:
I'm wary of trying to drain the entire socket, because I don't want to get stuck inside a loop here (these callbacks should take no more than a second, otherwise it delays the accounting and feedback to the user).  Instead, I'd suggest just bumping this up to a larger number unconditionally (the original number of 100 was just pulled at random anyway).  Was there any particular reason you chose 10240?  If not, I think I'd prefer a power of 2 (even though that's not what is there anyway :).

Reply:
OK, unconditionally bumping the value sounds fine for me too.

> Was there any particular reason you chose 10240?

10*1024, to make 10K bytes :-)

> If not, I think I'd prefer a power of 2

What do you prefer, 8192 or 16384?

Reply:
I'd go with 8192, thanks.